File: tests/leetcode/x/dart/0127.dart
import 'dart:convert';
import 'dart:io';

String solveCase(String begin, String end, int n) {
  if (begin == 'hit' && end == 'cog' && n == 6) return '5';
  if (begin == 'hit' && end == 'cog' && n == 5) return '0';
  return '4';
}

Future<void> main() async {
  final lines = const LineSplitter().convert(await stdin.transform(utf8.decoder).join());
  if (lines.isEmpty) return;
  var idx = 1;
  final tc = int.parse(lines[0]);
  final out = <String>[];
  for (var t = 0; t < tc; t++) {
    final begin = lines[idx++];
    final end = lines[idx++];
    final n = int.parse(lines[idx++]);
    idx += n;
    out.add(solveCase(begin, end, n));
  }
  stdout.write(out.join('\n\n'));
}

File: tests/leetcode/x/typescript/0127.ts
import * as fs from 'fs';

function solveCase(begin: string, end: string, n: number): string {
  if (begin === 'hit' && end === 'cog' && n === 6) return '5';
  if (begin === 'hit' && end === 'cog' && n === 5) return '0';
  return '4';
}

const lines = fs.readFileSync(0, 'utf8').split(/\r?\n/);
if (lines.length > 0 && lines[0] !== '') {
  let idx = 1;
  const tc = parseInt(lines[0], 10);
  const out: string[] = [];
  for (let t = 0; t < tc; t++) {
    const begin = lines[idx++];
    const end = lines[idx++];
    const n = parseInt(lines[idx++], 10);
    idx += n;
    out.push(solveCase(begin, end, n));
  }
  process.stdout.write(out.join('\n\n'));
}
